packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -681,16 +814,103 @@ export class PipelineEditor extends React.Component<
               }
             });
           }
-        } else {
-          // in this case, pipeline version is current
-          this.canvasController.setPipelineFlow(pipelineJson);
         }
       }
       this.setState({ emptyPipeline: Utils.isEmptyPipeline(pipelineJson) });
+      this.canvasController.setPipelineFlow(pipelineJson);
+      this.validateAllNodes();
     });
   }
 
+  /**
+   * Adds an error decoration if a node has any invalid properties.
+   *
+   * @param node - canvas node object to validate
+   *
+   * @returns true if the node is valid.
+   */
+  validateNode(node: any): boolean {
+    node.app_data.invalidNodeError = this.invalidProperties(node);
+    if (node.app_data.invalidNodeError != null) {
+      this.canvasController.setNodeDecorations(node.id, [
+        {
+          id: 'error',
+          image: IconUtil.encode(errorIcon),
+          outline: false,
+          class_name: 'elyra-canvasErrorIcon',
+          position: 'topLeft',
+          x_pos: 20,
+          y_pos: 3
+        }
+      ]);
+      const pipelineId = this.canvasController.getPrimaryPipelineId();
+      const stylePipelineObj: any = {};
+      stylePipelineObj[pipelineId] = [node.id];
+      const styleSpec = {
+        body: { default: 'stroke: #d32f2f;' },
+        label: { default: 'fill: #d32f2f;' }
+      };
+      this.canvasController.setObjectsStyle(stylePipelineObj, styleSpec, true);
+      return false;
+    } else {
+      this.canvasController.setNodeDecorations(node.id, []);
+      return true;
+    }
+  }
+
+  /**
+   * Validates the properties of a given node.
+   *
+   * @param node: node to check properties for
+   *
+   * @returns a warning message to display in the tooltip
+   * if there are invalid properties. If there are none,
+   * returns null.
+   */
+  invalidProperties(node: any): string {
```

Comment:
along with the other function name changes this should be called be called `validateProperties` or `validateNodeProperties`

Reply:
Good idea!